packages/phpstan-static-type-mapper/src/TypeMapper/IterableTypeMapper.php

```diff
@@ -87,17 +87,17 @@ private function convertUnionArrayTypeNodesToArrayTypeOfUnionTypeNodes(
         UnionTypeNode $unionTypeNode
     ): AttributeAwareUnionTypeNode {
         $unionedArrayType = [];
-        foreach ($unionTypeNode->types as $unionedType) {
-            if ($unionedType instanceof UnionTypeNode) {
```

Comment:
This special case should be kept, as `unionedType` states clearly that is iterated value. While `type` is just one main type

Reply:
Fixed by skip property fetch not from `ThisType`